tema.js

```diff
@@ -0,0 +1,164 @@
+'use strict'
+
+class erori extends Error {
+  static ERROR_CODE = {
+       METHOD_NOT_IMPLEMENTED: 'METHOD_NOT_IMPLEMENTED',
+       EXPIRED_PRODUCT: 'EXPIRED_PRODUCT',
+       NOT_ENOUGH_MONEY: 'NOT_ENOUGH_MONEY'
+   };
+
+   static ERROR_MESSAGE = {
+       [erori.ERROR_CODE.METHOD_NOT_IMPLEMENTED]: 'You must implement this method',
+       [erori.ERROR_CODE.EXPIRED_PRODUCT]: 'This product is expired',
+       [erori.ERROR_CODE.NOT_ENOUGH_MONEY]: "This customer doesn't have enough money"
+   };
+
+   constructor({ code, message = erori.ERROR_MESSAGE[code], data }) {
+       super(message);
+
+       this.code = code;
+       this.data = data;
+   }
+}
+
+
+class Person {
+  constructor() {}
+}
+
+class Customer extends Person {
+  #money
+  #shopping_cart
+
+  constructor (money, shopping_cart) {
+    super();
+    this.#money = money
+    this.#shopping_cart = shopping_cart
+  }
+
+  addToCart(product) {
+    try {
```

Comment:
Try to keep the logic simple. In this case it's enough to throw the error if the product is expired, otherwise to add it to the shopping cart.

The `try { } catch () { }` block makes sense only if the `.getDate()` call on the Product instance can throw, which is not the case.